seacatauth/middleware.py

```diff
@@ -27,14 +27,41 @@ async def auth_middleware(request, handler):
 		"""
 		Authenticate and authorize all incoming requests.
 		Raise HTTP 401 if authentication or authorization fails.
+
+		Metrics endpoint can be accessed with simple authorization using configured bearer token requesting the Private WebContainer directly.
+
+		Prometheus configuration example:
+		scrape_configs:
+			- job_name: 'seacat'
+				metrics_path: '/asab/v1/metrics'
+				scrape_interval: 10s
+				static_configs:
+				- targets: ['seacat-auth-svc:8082']
+				authorization:
+				- credentials: xtA4J9c6KK3g_Y0VplS_Rz4xmoVoU1QWrwz9CHz2p3aTpHzOkr0yp3xhcbkJK-ZO
 		"""
+		asab.Config.add_defaults(
+			{
+				"asab:metrics:auth": {
+					"bearer": "xtA4J9c6KK3g_Y0VplS_Rz4xmoVoU1QWrwz9CHz2p3aTpHzOkr0yp3xhcbkJK-ZO"
+				}
+			}
+		)
 		try:
 			# Authorize by OAuth Bearer token
 			# (Authorization by cookie is not allowed for API access)
 			request.Session = await oidc_service.get_session_from_authorization_header(request)
 		except KeyError:
 			request.Session = None
 
+		# Metrics
+		if (
+			request.path.endswith("/asab/v1/metrics")
+			and request.method == "GET"
+		):
+			if request.headers.get("Authorization") == asab.Config.get("asab:metrics:auth", "bearer"):
+				return await handler(request)
```

Comment:
Check this directly in the handler, not here .... you don't want to do this non-systematic `if`.

Raise `Unauthorized` exception then bearer token doesn't match.

Also, the check should be applied only if the bearer token is provided in the config - otherwise the endpoint should stay open.

Reply:
the handler is in asab... but I understand, I'll move it there...

Reply:
Yesterday, I removed everything from this middleware, there is no authorization, no session location for these requests.